### PR TITLE
add p10/12x

### PR DESCRIPTION
**Pull request by Benjamin-Freoua-Alma**
### Reason for change

[add p10/12x](https://linear.app/almapay/issue/MPP-555/add-p1012x)

### Code changes

Add 10 and 12 installment in select form

### How to test

Create and pay with 10 and 12 installment

### Checklist for authors and reviewers

- [ ] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [ ] The PR implements the changes asked in the referenced task / issue
- [ ] You understand the impact of this PR on existing code/features

### Non applicable

- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

**Review by Francois-Gomis — COMMENTED** [review]
Need frontend design for credit fees.

File: src/Controller/InstallmentPlanController.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+use Sylius\Component\Locale\Context\LocaleContextInterface;
+use Sylius\PayPalPlugin\Processor\LocaleProcessorInterface;
+use Sylius\PayPalPlugin\Provider\AvailableCountriesProviderInterface;
+use Sylius\PayPalPlugin\Provider\PayPalConfigurationProviderInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Twig\Environment;
+
+final class InstallmentPlanController
```
Comment (ClaireAlma) [inline src/Controller/InstallmentPlanController.php]: mettre un commentaire de classe

File: src/Controller/InstallmentPlanController.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+{
+    private Environment $twig;
```
Comment (ClaireAlma) [inline src/Controller/InstallmentPlanController.php]: mettre les commentaires des variables de classe

File: src/Controller/InstallmentPlanController.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    private LocaleContextInterface $localeContext;
+
+    private OrderRepositoryInterface $orderRepository;
+
+    private AvailableCountriesProviderInterface $availableCountriesProvider;
+
+    private LocaleProcessorInterface $localeProcessor;
+
+    public function __construct(
```
Comment (ClaireAlma) [inline src/Controller/InstallmentPlanController.php]: Manque le commentaire de fonction

File: src/Controller/InstallmentPlanController.php [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        $this->twig = $twig;
+        $this->router = $router;
+        $this->channelContext = $channelContext;
+        $this->localeContext = $localeContext;
+        $this->orderRepository = $orderRepository;
+        $this->availableCountriesProvider = $availableCountriesProvider;
+        $this->localeProcessor = $localeProcessor;
+    }
+
+    public function renderPaymentPageInstallmentPlanAction(Request $request): Response
```
Comment (ClaireAlma) [inline src/Controller/InstallmentPlanController.php]: commentaire de fonction

File: src/Controller/InstallmentPlanController.php [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    {
+        $orderId = $request->attributes->getInt('orderId');
+        /** @var ChannelInterface $channel */
+        $channel = $this->channelContext->getChannel();
+        /** @var OrderInterface $order */
+        $order = $this->orderRepository->find($orderId);
+
+        try {
```
Comment (ClaireAlma) [inline src/Controller/InstallmentPlanController.php]: remontrer le try pour englober le debut du code dans la fonction

File: src/Controller/InstallmentPlanController.php [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+                'currency' => $order->getCurrencyCode(),
+                'completePayPalOrderFromPaymentPageUrl' => $this->router->generate('sylius_paypal_plugin_complete_paypal_order_from_payment_page', ['id' => $orderId]),
+                'createPayPalOrderFromPaymentPageUrl' => $this->router->generate('sylius_paypal_plugin_create_paypal_order_from_payment_page', ['id' => $orderId]),
+                'errorPayPalPaymentUrl' => $this->router->generate('sylius_paypal_plugin_payment_error'),
+                'locale' => $this->localeProcessor->process((string) $order->getLocaleCode()),
+                'orderId' => $orderId,
+                'partnerAttributionId' => $this->payPalConfigurationProvider->getPartnerAttributionId($channel),
+            ]));
+        } catch (\InvalidArgumentException $exception) {
+            return new Response('');
```
Comment (ClaireAlma) [inline src/Controller/InstallmentPlanController.php]: logger les exceptions,et dans un controller on doit catcher tous les types d'exception possible pas que InvalidArgumentException

File: src/Form/Type/SelectPaymentType.php [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+declare(strict_types=1);
+
+namespace Alma\SyliusPaymentPlugin\Form\Type;
+
+use Sylius\Bundle\CoreBundle\Form\Type\Checkout\PaymentType;
+use Sylius\Bundle\ResourceBundle\Form\Type\AbstractResourceType;
+use Symfony\Component\Form\FormBuilderInterface;
+
+final class SelectPaymentType extends AbstractResourceType
```
Comment (ClaireAlma) [inline src/Form/Type/SelectPaymentType.php]: manque commentaires de class et fonction

File: src/Form/Type/SelectPaymentType.php [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        $builder->add('payments', ChangePaymentMethodType::class, [
+            'entry_type' => PaymentType::class,
+            'label' => false,
+        ]);
+    }
+
+    public function getBlockPrefix(): string
+    {
+        return 'sylius_checkout_select_payment';
```
Comment (ClaireAlma) [inline src/Form/Type/SelectPaymentType.php]: passer cette chaine de caractere en constante

**Review by ClaireAlma — CHANGES_REQUESTED** [review]

**Comment by ClaireAlma** [comment]
@Benjamin-Freoua-Alma Miss the changelog informations

File: src/Controller/InstallmentsController.php [hunk 9]
```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Alma\SyliusPaymentPlugin\Controller;
+
+use Alma\SyliusPaymentPlugin\Bridge\AlmaBridgeInterface;
```
Comment (joyet-simon) [inline src/Controller/InstallmentsController.php]: This import is unused

File: src/Resources/views/_partials/_javascript.html.twig [hunk 10]
```diff
@@ -0,0 +1,2 @@
+<script src="{{ asset('/bundles/almasyliuspaymentplugin/js/alma.js') }}"></script>
+ClaireTest
```
Comment (joyet-simon) [inline src/Resources/views/_partials/_javascript.html.twig]: To remove

File: src/Resources/views/_partials/_almaPnxPlus4.html.twig
Comment (joyet-simon) [inline src/Resources/views/_partials/_almaPnxPlus4.html.twig:1]: Maybe rename to _almaCredit.html.twig

File: src/Resources/public/js/alma.js [hunk 11]
```diff
@@ -0,0 +1,8 @@
+
+$(function() {
+    console.log('test cclaire !!!!');
+    $('input[data-payment*=alma-method]').on('click', function() {
+        console.log('click click !!!!');
```
Comment (joyet-simon) [inline src/Resources/public/js/alma.js]: Remove console logs

**Review by joyet-simon — APPROVED** [review]

**Comment by ClaireAlma** [comment]
@joyet-simon tu as tester ? on ne valide plus les PR si on ne les testes pas ou sinon on écrit en commentaire, review ok mais non testée. Merci

File: src/Resources/views/bundles/SyliusShopBundle/Checkout/SelectPayment/_choiceAlma.html.twig [hunk 12]
```diff
@@ -0,0 +1,21 @@
+<div class="item" {{ sylius_test_html_attribute('payment-item') }}>
+    <div class="field">
+        <div class="ui radio checkbox" {{ sylius_test_html_attribute('payment-method-checkbox') }}>
+            {{ form_widget(form, sylius_test_form_attribute('payment-method-select')|sylius_merge_recursive( {'attr': {'data-payment': 'alma-method-id-' ~ method.id}})) }}
+        </div>
+    </div>
+    <div class="content">
+        <a class="header">
+            {{ form_label(form, null, {'label_attr': {'data-test-payment-method-label': ''}}) }}
+            <img src="{{ asset('/bundles/almasyliuspaymentplugin/img/p' ~ method.gatewayConfig.config.installments_count ~ 'x_logo.svg') }}" class="ui center aligned inline image" alt="Logo Alma" />
```
Comment (ClaireAlma) [inline src/Resources/views/bundles/SyliusShopBundle/Checkout/SelectPayment/_choiceAlma.html.twig]: Alma logo is not translated

**Review by ClaireAlma — APPROVED** [review]

File: composer.json [hunk 13]
```diff
@@ -13,9 +13,9 @@
   "description": "Payment plugin to integrate Alma installments and pay later payments with your Sylius shop",
   "license": "MIT",
   "require": {
-    "php": ">=7.1",
-    "alma/alma-php-client": "^1.11.1",
-    "sylius/sylius": ">=1.6.9",
+    "php":  "~7.3 || ~7.4 || ~8.0 || ~8.1 || ~8.2",
+    "alma/alma-php-client": ">=1.11.1",
```
Comment (ClaireAlma) [inline composer.json]: il faudra pas oublier de mettre la version du PHP client

File: src/Helper/EligibilityHelper.php [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+            ],
+            "billing_address" => [
+                "country" => $billingCountryCode,
+            ],
+            "shipping_address" => [
+                "country" => $shippingCountryCode
+            ],
+            "locale" => $locale,
+        ];
+
```
Comment (Francois-Gomis) [inline src/Helper/EligibilityHelper.php]: you can split data builder and eligibility call

File: src/Resources/translations/flashes.de.yml [hunk 15]
```diff
@@ -0,0 +1,5 @@
+alma_sylius_payment_plugin:
```
Comment (Francois-Gomis) [inline src/Resources/translations/flashes.de.yml:1]: No deepl translation ?

**Review by Francois-Gomis — APPROVED** [review]

File: src/Helper/EligibilityHelper.php (hunk 14 above)
Comment (Benjamin-Freoua-Alma) [reply src/Helper/EligibilityHelper.php]: @Francois-Gomis 
I added the eligibilityDataBuilder :)

**Pull request merged**